src/streamlink/plugins/twitch.py

```diff
@@ -697,12 +703,19 @@ def _check_for_host(self):
             self.logger.info("{0} is hosting {1}".format(self.channel, host_info["target_login"]))
             return host_info["target_login"]
 
+    def _check_for_rerun(self):
+        return self.api.streams(self.channel_id)["stream"]["stream_type"] == "rerun"
+
     def _get_hls_streams(self, stream_type="live"):
         self.logger.debug("Getting {0} HLS streams for {1}".format(stream_type, self.channel))
         self._authenticate()
         self._hosted_chain.append(self.channel)
 
         if stream_type == "live":
+            if self._check_for_rerun() and self.options.get("disable_reruns"):
```

Comment:
This will make an API call even if reruns are not disabled, which is unnecessary and will delay launching the stream by a bit.

Reply:
Bah my bad @bastimeyer I totally forgot that it would cause an API call.